src/ucp/core/ucp_worker.c

```diff
@@ -487,6 +487,15 @@ static unsigned ucp_worker_iface_err_handle_progress(void *arg)
                                   UCT_FLUSH_FLAG_CANCEL,
                                   ucp_ep_err_pending_purge,
                                   UCS_STATUS_PTR(status));
+        /* UCT CM lane mustn't be scheduled on worker progress when discarding,
+         * since UCP EP will be destroyed due to peer failure and
+         * ucp_cm_disconnect_cb() could be invoked on async thread after UCP EP
+         * is destroyed and before UCT CM EP is destroyed from discarding
+         * functionality. So, UCP EP will passed as a corrupted argument to
+         * ucp_cm_disconnect_cb() */
+        ucs_assert((lane != ucp_ep_get_cm_lane(ucp_ep)) ||
+                   !ucp_worker_is_uct_ep_discarding(worker,
+                                                    ucp_ep->uct_eps[lane]));
```

Comment:
this assert is good, i would write it like this to be more readable:
```
if (lane == ucp_ep_get_cm_lane(ucp_ep) {
  ucs_assert(!discarding)
}
```

Reply:
done